core/dbt/task/compile.py

```diff
@@ -129,14 +134,26 @@ def defer_to_manifest(self, adapter, selected_uids: AbstractSet[str]):
 
     def _runtime_initialize(self):
         if getattr(self.args, "inline", None):
-            block_parser = SqlBlockParser(
-                project=self.config, manifest=self.manifest, root_project=self.config
-            )
-            sql_node = block_parser.parse_remote(self.args.inline, "inline_query")
-            process_node(self.config, self.manifest, sql_node)
-            # keep track of the node added to the manifest
-            self._inline_node_id = sql_node.unique_id
-
+            try:
+                block_parser = SqlBlockParser(
+                    project=self.config, manifest=self.manifest, root_project=self.config
+                )
+                sql_node = block_parser.parse_remote(self.args.inline, "inline_query")
+                process_node(self.config, self.manifest, sql_node)
+                # keep track of the node added to the manifest
+                self._inline_node_id = sql_node.unique_id
+            except CompilationError as exc:
+                fire_event(
+                    ParseNodeError(
```

Comment:
I made this request before, but somehow github lost it... "ParseNodeError" is too generic of a name, it sounds like it might apply to any parsing error, but it's only fired for inline nodes. Could we call it something like "ParseInlineNodeError"?

Reply:
The reason I didn't name this to be specific for `inline node` is the same error can happened for any node.
Given the exception I am catching here, do you think `NodeCompilationError` would be a better one that's not too generic? Or any other suggestions that doesn't contain `inline node`?

Reply:
Per offline discussion with Gerda, we decided that raising a narrower scoped exception is fine because distinguishing between inline node parsing and project node parsing might be useful.